Clarify how certificate fingerprints work for Fleet Server (#5138)
elastic/docs-content: reference/fleet/certificate-fingerprints.md

::::{important}
The certificate whose fingerprint you configure must be present in the certificate chain sent by the server during the TLS handshake. If the certificate is not in the chain, the fingerprint cannot be matched, and the connection will fail with a `certificate signed by unknown authority` error.
::::

### CA certificate method

When using `ssl.certificate_authorities` in the configuration or the `--fleet-server-es-ca` CLI option:

1. The server presents its certificate chain during the TLS handshake.
2. The client uses the CA certificate file to validate the chain.
3. The root CA does not need to be in the server's presented chain because the client already has it locally.

## Requirements [fingerprint-requirements]

For a certificate fingerprint to work correctly, the certificate and fingerprint must meet these requirements:

* [The certificate must be in the server's presented chain](#fingerprint-requirements-1)
* [The certificate must be a CA certificate](#fingerprint-requirements-2)
* [The fingerprint must be correctly formatted](#fingerprint-requirements-3)

### The certificate must be in the server's presented chain [fingerprint-requirements-1]

> The certificate must be included in the certificate file that the server presents during the TLS handshake. For {{es}}, this is the file specified in the `xpack.security.http.ssl.certificate` setting. For {{fleet-server}}, this is the certificate specified in the {{fleet-server}} configuration.

Reviewer: Not sure if we need `{{fleet-server}}` twice in this sentence? 
```suggestion
The certificate must be included in the certificate file that the server presents during the TLS handshake. For {{es}}, this is the file specified in the `xpack.security.http.ssl.certificate` setting. For {{fleet-server}}, this is the certificate specified in its configuration.
```

for fingerprints because they are not sent during the TLS handshake.

For a practical example of how this works with a real certificate chain, refer to [Choosing which certificate to use](#choosing-certificate).

### The certificate must be a CA certificate [fingerprint-requirements-2]

The certificate must have `CA:TRUE` in its X509v3 Basic Constraints. Server certificates cannot be used with the fingerprint method.

To check if a certificate is a CA certificate, use:

```bash
openssl x509 -noout -text -in certificate.crt
```

Look for this section in the output:

```text
X509v3 extensions:
    X509v3 Basic Constraints: critical
        CA:TRUE
```

If `CA:TRUE` is present, the certificate can be used with the fingerprint method.

### The fingerprint must be correctly formatted [fingerprint-requirements-3]

The fingerprint must be a HEX-encoded SHA-256 hash with colons removed.

## Choosing which certificate to use [choosing-certificate]

Before generating a fingerprint, you need to identify which certificate from your certificate 
chain to use. You must use a CA certificate from the certificate file that the server presents 

> during the TLS handshake, and not from the server's certificate authorities file.

Reviewer: ```suggestion
Before generating a fingerprint, you need to identify which certificate from your certificate 
chain to use. You must use a CA certificate from the certificate file the server presents 
during the TLS handshake, not from the server's certificate authorities file.
```

### Understanding your certificate chain

Consider this certificate chain as an example:

```text
Root CA → Intermediate CA 1 → Intermediate CA 2 → Server Certificate
```

Let's assume that this chain is split across two certificate files in the {{es}} configuration:

```yaml
xpack.security.http.ssl.certificate: certs/server-intermediate2-intermediate1.pem
xpack.security.http.ssl.certificate_authorities: ["certs/root-ca.pem"]
```

:::{note}

> This example shows Elasticsearch configuration settings. If you're connecting to Fleet Server, the 

Reviewer: ```suggestion
This example shows {{es}} configuration settings. If you're connecting to {{fleet-server}}, the 
```
